src/toil/cwl/cwltoil.py

```diff
@@ -193,6 +193,78 @@ class SkipNull:
     """
 
 
+def trim_mounts_op_down(file_or_directory: CWLObjectType) -> None:
+    return
+
+
+def sniff_location(file_or_directory: CWLObjectType) -> Optional[str]:
+    """
+    Return None if we don't have a local path or file URI
+    :param file_or_directory:
+    :return:
+    """
+    # Since we only consider mountable paths, if path is not file URI or bare path, don't consider it
+    path = cast(str, file_or_directory.get('location') or file_or_directory.get('path'))
+    scheme = urlparse(path).scheme
+    if scheme == '' or scheme == 'file':
+        return path
+    else:
+        return None
+
+
+def trim_mounts_op_up(file_or_directory: CWLObjectType, op_down_ret: None, child_results: list[bool]) -> bool:
+    """
+    Nonredundant for something in a directory means its path or location is not within the parent directory or doesn't match its basename
```

Comment:
The first line in the docstring should summarize the function.

```suggestion
    Remove subtrees of the CWL file or directory object tree that only have redundant stuff in them.

    Nonredundant for something in a directory means its path or location is not within the parent directory or doesn't match its basename
```